packages/block/src/header-from-rpc.ts

```diff
@@ -30,10 +30,10 @@ export default function blockHeaderFromRpc(blockParams: any, options?: BlockOpti
     options,
   )
 
-  // override hash in case something was missing
-  blockHeader.hash = function () {
-    return toBuffer(blockParams.hash)
-  }
+  // override hash in case something was missing TODO: why do we need this?
+  //blockHeader.hash = function () {
+  //  return toBuffer(blockParams.hash)
+  //}
```

Comment:
Why do we have this? I don't see any use case to override the hash function?

Reply:
I believe the code was previously used to import incomplete block data from RPC but had the hash provided as a primary identifier so in this way it was always being used.

I just checked and we're only using it for our benchmarks now, and there we have complete block data (+txs) - so we can nuke it here, but I think we should note if we drop support.

Since this is confusing we can add some documentation or some test cases of e.g. rpc block data without tx data.

Reply:
actually my bad, I now realize that the txs don't factor into the hash, just header components.

i think this was here to protect the integrity of a passed-in hash if the other elements don't add up (as per the comment "override hash in case something was missing"), but i don't see a need for this for us anymore so i will remove it for now.

in that same spirit i also don't see the need for the from-rpc faketx either, since it's a similar pattern, so i will also remove that.

i'm not sure what use cases we had prior where these were necessary, but there is nothing in our code or test suites that rely on them anymore.